src/phasorpy/phasor.py

```diff
@@ -34,23 +35,216 @@
-    from ._typing import Any, NDArray, ArrayLike, Literal
-
-import math
+    from ._typing import Any, NDArray, ArrayLike, DTypeLike, Literal
 
 import numpy
 
+from ._phasor import _phasor_from_lifetime, _phasor_from_signal
+
+
+def phasor_from_signal_f1(
+    signal: ArrayLike,
+    /,
+    *,
+    axis: int = 0,
+    sample_phase: ArrayLike | None = None,
+    dtype: DTypeLike = None,
+    num_threads: int | None = None,
+) -> tuple[NDArray[Any], NDArray[Any], NDArray[Any]]:
+    """Return phasor coordinates of fundamental frequency from signal.
+
+    Compared to the :py:func:`phasor_from_signal` reference implementation,
+    this function does not use FFT, considers only the fundamental frequency
+    (first harmonic), uses less memory, might be faster, and supports
+    out-of-order samples.
+
+    Parameters
+    ----------
+    signal : array_like
+        Frequency-domain, time-domain, or hyperspectral data.
+        A minimum of three samples are required along `axis`.
+        The samples must be uniformly spaced, but not necessarily in order.
+        Must be of type uint8, uint16, uint32, uint64, int32, float32, or
+        float64.
+    axis : int, optional
+        Axis over which to compute phasor coordinates.
+        The default is the first axis (0).
+    sample_phase : array_like, optional
+        Phase values (in radians) of `signal` samples along `axis`.
+        If None (default), samples are assumed to be uniformly spaced along
+        one period.
+        The array size must equal the number of samples along `axis`.
+    dtype : dtype_like, optional
+        Data type of output arrays. Either float32 or float64 (default).
+    num_threads : int, optional
+        Number of OpenMP threads to use for parallelization.
+        By default, multi-threading is disabled.
+        If zero, up to half of logical CPUs are used.
+        OpenMP may not be available on all platforms.
+
+    Returns
+    -------
+    mean : ndarray
+        Average of signal along axis.
+    real : ndarray
+        Real component of phasor coordinates along axis.
+    imag : ndarray
+        Imaginary component of phasor coordinates along axis.
+
+    Raises
+    ------
+    ValueError
+        The `signal` has less than three samples along `axis` or the
+        `sample_phase` size does not equal the number of samples along `axis`.
+    TypeError
+        The `signal` or `dtype` types are not supported.
+
+    Examples
+    --------
+    Calculate phasor coordinates of a phase-shifted sinusoidal waveform:
+
+    >>> sample_phase = numpy.linspace(0, 2 * math.pi, 5, endpoint=False)[::-1]
+    >>> signal = 1.1 * (
+    ...     numpy.cos(sample_phase - 0.78539816) * 2 *  0.70710678 + 1
+    ... )
+    >>> phasor_from_signal_f1(
+    ...    signal, sample_phase=sample_phase
+    ... )  # doctest: +NUMBER
+    (1.1, 0.5, 0.5)
+
+    """
+    signal = numpy.array(signal, order='C', ndmin=1, copy=False)
+    # assert signal.flags['C_CONTIGUOUS']
```

Comment:
This should be taken care of by `order='C'` in the line before. I just kept this as a comment/reminder in case the previous line changes. I will double-check that there's a test case.

Reply:
I added a test for a non-contiguous signal array.